core/src/main/java/edu/ucr/cs/riple/core/Config.java

```diff
@@ -278,6 +333,22 @@ private static void showHelpAndQuit(HelpFormatter formatter, Options options) {
     System.exit(1);
   }
 
+  /**
+   * Returns the set of downstream dependencies, returns empty list if {@link
+   * Config#downStreamDependenciesAnalysisActivated} is false.
+   *
+   * @return ImmutableSet of {@link edu.ucr.cs.riple.core.metadata.submodules.Module}.
+   */
+  public ImmutableSet<Module> getSubModules() {
+    if (!this.downStreamDependenciesAnalysisActivated) {
+      return ImmutableSet.of();
+    }
+    final int[] id = {1};
+    return downstreamModulesBuildCommands.stream()
+        .map(command -> new Module("module-" + id[0]++, command))
```

Comment:
is this incrementing the value of `id[0]` as a side effect ? Maybe `for loops` are better in such scenarios. 

OR 
```
IntStream.range(0, downstreamModulesBuildCommands.size())
.map(i -> new Module("module-", + {i} + downstreamModulesBuildCommand.get(i)))
.collect(...)
```

Reply:
Yes it is incrementing as a side effect. Integers cant be used from outer scope unless they are `final` therefore needs to be wrapped to an array. Also `downstreamModulesBuildCommands` is an `ImmutableSet` and it does not have a `get()` method and I can't write it as for loop.